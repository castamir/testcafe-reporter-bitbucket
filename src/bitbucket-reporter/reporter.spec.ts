import normalizeNewline from "normalize-newline";

import createReport from "../test/create-report";

describe("bitbucket reporter", () => {
  it("should produce report with colors", function() {
    let report = createReport(true);
    report = normalizeNewline(report).trim();

    expect(report).toMatchSnapshot();
  });

  it("should produce report without colors", function() {
    let report = createReport(false);
    report = normalizeNewline(report).trim();

    expect(report).toMatchSnapshot();
  });
});
