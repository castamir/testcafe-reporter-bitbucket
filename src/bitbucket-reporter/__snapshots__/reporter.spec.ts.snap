// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bitbucket reporter should produce report with colors 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<testsuite name=\\"TestCafe Tests: Chrome 41.0.2227 / Mac OS X 10.10.1, Firefox 47 / Mac OS X 10.10.1\\" tests=\\"6\\" failures=\\"2\\" skipped=\\"1\\" errors=\\"0\\" time=\\"925\\" timestamp=\\"Thu, 01 Jan 1970 00:15:25 GMT\\" >
  <testcase classname=\\"First fixture\\" name=\\"First test in first fixture (unstable) (screenshots: /screenshots/1445437598847)\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"First fixture\\" name=\\"Second test in first fixture (screenshots: /screenshots/1445437598847)\\" time=\\"74\\">
    <failure>
    <![CDATA[
      1) Error on page \\"http://example.org\\":

         Some error

         Browser: Chrome 41.0.2227 / Mac OS X 10.10.1
         Screenshot: /screenshots/1445437598847/errors

            1 |import createCallSiteRecord from \\"callsite-record\\";
            2 |
            3 |function someFunc() {
          > 4 |  throw new Error(\\"Hey ya!\\");
            5 |}
            6 |
            7 |let result;
            8 |
            9 |try {

            at someFunc (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at jasmine2 (some-file:1:1)
            at runTestInternal (some-file:1:1)

      2) The specified selector does not match any element in the DOM tree.

         Browser: Firefox 47 / Mac OS X 10.10.1

            1 |import createCallSiteRecord from \\"callsite-record\\";
            2 |
            3 |function someFunc() {
          > 4 |  throw new Error(\\"Hey ya!\\");
            5 |}
            6 |
            7 |let result;
            8 |
            9 |try {

            at someFunc (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at jasmine2 (some-file:1:1)
            at runTestInternal (some-file:1:1)
    ]]>
    </failure>
  </testcase>
  <testcase classname=\\"First fixture\\" name=\\"Third test in first fixture\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"Second fixture\\" name=\\"First test in second fixture\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"Second fixture\\" name=\\"Second test in second fixture\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"Second fixture\\" name=\\"Third test in second fixture\\" time=\\"0\\">
    <skipped/>
  </testcase>
  <testcase classname=\\"Third fixture\\" name=\\"First test in third fixture (unstable)\\" time=\\"74\\">
    <failure>
    <![CDATA[
      1) - Error in beforeEach hook -
         The specified selector does not match any element in the DOM tree.

         Browser: Firefox 47 / Mac OS X 10.10.1

            1 |import createCallSiteRecord from \\"callsite-record\\";
            2 |
            3 |function someFunc() {
          > 4 |  throw new Error(\\"Hey ya!\\");
            5 |}
            6 |
            7 |let result;
            8 |
            9 |try {

            at someFunc (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at jasmine2 (some-file:1:1)
            at runTestInternal (some-file:1:1)
    ]]>
    </failure>
  </testcase>
  <system-out>
  <![CDATA[
    Warnings (3):
    --
      Was unable to take a screenshot due to an error.

      ReferenceError: someVar is not defined
    --
      Was unable to take a screenshot due to an error.

      ReferenceError: someOtherVar is not defined
    --
      Was unable to take screenshots because the screenshot directory is not specified. To specify it, use the \\"-s\\" or \\"--screenshots\\" command line option or the \\"screenshots\\" method of the test runner in case you are using API.
  ]]>
  </system-out>
</testsuite>"
`;

exports[`bitbucket reporter should produce report without colors 1`] = `
"<?xml version=\\"1.0\\" encoding=\\"UTF-8\\" ?>
<testsuite name=\\"TestCafe Tests: Chrome 41.0.2227 / Mac OS X 10.10.1, Firefox 47 / Mac OS X 10.10.1\\" tests=\\"6\\" failures=\\"2\\" skipped=\\"1\\" errors=\\"0\\" time=\\"925\\" timestamp=\\"Thu, 01 Jan 1970 00:15:25 GMT\\" >
  <testcase classname=\\"First fixture\\" name=\\"First test in first fixture (unstable) (screenshots: /screenshots/1445437598847)\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"First fixture\\" name=\\"Second test in first fixture (screenshots: /screenshots/1445437598847)\\" time=\\"74\\">
    <failure>
    <![CDATA[
      1) Error on page \\"http://example.org\\":

         Some error

         Browser: Chrome 41.0.2227 / Mac OS X 10.10.1
         Screenshot: /screenshots/1445437598847/errors

            1 |import createCallSiteRecord from \\"callsite-record\\";
            2 |
            3 |function someFunc() {
          > 4 |  throw new Error(\\"Hey ya!\\");
            5 |}
            6 |
            7 |let result;
            8 |
            9 |try {

            at someFunc (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at jasmine2 (some-file:1:1)
            at runTestInternal (some-file:1:1)

      2) The specified selector does not match any element in the DOM tree.

         Browser: Firefox 47 / Mac OS X 10.10.1

            1 |import createCallSiteRecord from \\"callsite-record\\";
            2 |
            3 |function someFunc() {
          > 4 |  throw new Error(\\"Hey ya!\\");
            5 |}
            6 |
            7 |let result;
            8 |
            9 |try {

            at someFunc (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at jasmine2 (some-file:1:1)
            at runTestInternal (some-file:1:1)
    ]]>
    </failure>
  </testcase>
  <testcase classname=\\"First fixture\\" name=\\"Third test in first fixture\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"Second fixture\\" name=\\"First test in second fixture\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"Second fixture\\" name=\\"Second test in second fixture\\" time=\\"74\\">
  </testcase>
  <testcase classname=\\"Second fixture\\" name=\\"Third test in second fixture\\" time=\\"0\\">
    <skipped/>
  </testcase>
  <testcase classname=\\"Third fixture\\" name=\\"First test in third fixture (unstable)\\" time=\\"74\\">
    <failure>
    <![CDATA[
      1) - Error in beforeEach hook -
         The specified selector does not match any element in the DOM tree.

         Browser: Firefox 47 / Mac OS X 10.10.1

            1 |import createCallSiteRecord from \\"callsite-record\\";
            2 |
            3 |function someFunc() {
          > 4 |  throw new Error(\\"Hey ya!\\");
            5 |}
            6 |
            7 |let result;
            8 |
            9 |try {

            at someFunc (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at Runtime.requireModuleOrMock (some-file:1:1)
            at Object.<anonymous> (some-file:1:1)
            at Runtime._execModule (some-file:1:1)
            at Runtime._loadModule (some-file:1:1)
            at Runtime.requireModule (some-file:1:1)
            at jasmine2 (some-file:1:1)
            at runTestInternal (some-file:1:1)
    ]]>
    </failure>
  </testcase>
  <system-out>
  <![CDATA[
    Warnings (3):
    --
      Was unable to take a screenshot due to an error.

      ReferenceError: someVar is not defined
    --
      Was unable to take a screenshot due to an error.

      ReferenceError: someOtherVar is not defined
    --
      Was unable to take screenshots because the screenshot directory is not specified. To specify it, use the \\"-s\\" or \\"--screenshots\\" command line option or the \\"screenshots\\" method of the test runner in case you are using API.
  ]]>
  </system-out>
</testsuite>"
`;
